iceoryx_binding_c/include/iceoryx_binding_c/publisher.h

```diff
@@ -84,35 +84,37 @@ void iox_pub_deinit(iox_pub_t const self);
 
 /// @brief allocates a chunk in the shared memory
 /// @param[in] self handle of the publisher
-/// @param[in] chunk pointer in which a pointer to the allocated chunk is stored
-/// @param[in] payloadSize user payload size of the allocated chunk
+/// @param[in] userPayload pointer in which a pointer to the user-payload of the allocated chunk is stored
+/// @param[in] userPayloadSize user-payload size of the allocated chunk
 /// @return on success it returns AllocationResult_SUCCESS otherwise a value which
 ///         describes the error
-/// @note for the payload alignment `IOX_C_CHUNK_DEFAULT_USER_PAYLOAD_ALIGNMENT` is used
-///       for a custom payload alignment please use `iox_pub_loan_aligned_chunk`
-ENUM iox_AllocationResult iox_pub_loan_chunk(iox_pub_t const self, void** const chunk, const uint32_t payloadSize);
+/// @note for the user-payload alignment `IOX_C_CHUNK_DEFAULT_USER_PAYLOAD_ALIGNMENT` is used
+///       for a custom user-payload alignment please use `iox_pub_loan_aligned_chunk`
+ENUM iox_AllocationResult iox_pub_loan_chunk(iox_pub_t const self,
+                                             void** const userPayload,
```

Comment:
I resolved all the other comments regarding the naming to keep the discussion in this thread. Will adjust the code according the outcome of the discussion.

I also thought about that and initially didn't change it. But then I noticed that we already have inconsistencies because of this, e.g. `iox_chunk_payload_to_header`, `iox_sub_take_chunk` and `SampleDeleter::`operator() want payload pointer and there is the `ChunkHeader::fromPayload` static function which is used in the untyped API to get a pointer to the `ChunkHeader` from the payload pointer. Additionally, many parameter in lambdas were called payload. This indicates that something with our terminology is wrong or wrongly used.

Furthermore, with the user-header, the memory pools are configured with the size of the chunk-payload and the user-header and user-payload are part of the chunk-payload, so we need a clear distinction for this and this is also something a user must be aware of.

I came to the conclusion that a user loans a chunk, but what she get is a pointer to the user-payload, not the chunk itself or the chunk-header. Let's be clear what we provide to our users. It's the same with publish or release. A chunk is published or released, but this is done by providing a pointer to the user-payload of the chunk, not something else like a pointer to the actual chunk. I also don't think it's a good idea to call it chunk and then stating in the documentation that it's actually a user-payload. Just give it the name what it actually is.

Also, having an example with a user-header with different naming results in even more confusion and is the opposite of consistency.

Of course we need to update our examples and I would also volunteer to do that. I probably won't happen before the RC, though.

@budrus @MatthiasKillat @elfenpiff what do you think?

Reply:
I agree with userPayload as part of the chunk, consisting of: internal header, user header and user payload. What we get is the pointer to the payload part which can be used to get the internal header via other API functions. It is unfortunate that the method is called iox_pub_loan_chunk, could very well be iox_pub_loan. However this must be consistent with release on publisher and also on subscriber side, so probably not something for now.

And as you write, you loan a chunk, but get a pointer to a specific part of it, the user payload which is best reflected best like this. Long term I would remove the chunk part from the function names in the user API, but this requires discussion and should not happen this release.

Reply:
How about calling the parameter `usrePayloadOfChunk`? This should made it clear that it is a part of the chunk.

In the lambdas of the typed API the parameter will be called `sample`. It was a mistake to call it `payload` since the terminology in the typed API is `Sample`

Regarding the removal of `chunk` from the function names. As discussed with @mossmaurice, it would probably be better to keep it and converge the untyped C++ API with the C API by adding `Chunk` to `loan` and `release`. This would make it also more consistent with `loadPreviousChunk`

Reply:
okay, I changed the name but while changing it I noticed that I still prefer `userPayload` in the signature because it is more consistent with e.g. `userPayloadSize`. Since this is not part of the signature, it could be changed to just `userPayload` after the release and doesn't need to block the PR